### PR TITLE
Feature: add ios implementation

### PR DESCRIPTION
**Pull request by jonyas**
- Add ios implementation

### CODE REVIEW

File: ios/Classes/SwiftSegmentFlutterSdkPlugin.swift [hunk 1]
```diff
@@ -11,16 +12,83 @@ public class SwiftSegmentFlutterSdkPlugin: NSObject, FlutterPlugin {
...
+      result(FlutterMethodNotImplemented)
     }
   }
+
+  private func handleInitAnalytics(call: FlutterMethodCall, result: @escaping FlutterResult) {
+    guard let args = call.arguments as? [String: Any] else {
+      result(FlutterError(code: "ERROR", message: "Arguments are missing", details: nil))
+      return
+    }
+    guard let name = args["writeKey"] as? String else {
```
Comment (sdsykes) [inline ios/Classes/SwiftSegmentFlutterSdkPlugin.swift]: name should be writeKey - this shouldn't compile?

File: ios/Classes/SwiftSegmentFlutterSdkPlugin.swift [hunk 2]
```diff
@@ -11,16 +12,83 @@ public class SwiftSegmentFlutterSdkPlugin: NSObject, FlutterPlugin {
...
+    if let flushQueueSize = args["flushQueueSize"] as? Int {
+      config.flushAt = flushQueueSize
+    }
+    if let trackApplicationLifecycleEvents = args["trackApplicationLifecycleEvents"] as? Bool {
+      config.trackApplicationLifecycleEvents = trackApplicationLifecycleEvents
+    }
+    if let recordScreenViews = args["recordScreenViews"] as? Bool {
+      config.recordScreenViews = recordScreenViews
+    }
+    if let recordScreenViews = args["trackAttributionInformation"] as? Bool {
```
Comment (sdsykes) [inline ios/Classes/SwiftSegmentFlutterSdkPlugin.swift]: fix variable name

File: ios/Classes/SwiftSegmentFlutterSdkPlugin.swift [hunk 3]
```diff
@@ -11,16 +12,83 @@ public class SwiftSegmentFlutterSdkPlugin: NSObject, FlutterPlugin {
...
+      config.trackAttributionData = trackAttributionInformation
+    }
+    if let recordScreenViews = args["trackAttributionInformation"] as? Bool {
```
Comment (sdsykes) [inline ios/Classes/SwiftSegmentFlutterSdkPlugin.swift]: repeated

**Comment by jonyas** [comment]
All fixed!

**Review by sdsykes — APPROVED** [review]

**Pull request merged**